src/rgw/rgw_data_sync.cc

```diff
@@ -1057,7 +1057,7 @@ class RGWDataSyncSingleEntryCR : public RGWCoroutine {
         sync_env->observer->on_bucket_changed(bs.bucket.get_key());
       }
       /* FIXME: what do do in case of error */
-      if (marker_tracker && !entry_marker.empty()) {
+      if (marker_tracker && !entry_marker.empty() && sync_status == 0) {
```

Comment:
i think we do still want to update the marker here, so that an error from one bucket shard doesn't prevent progress on bucket shards that come later in the datalog

instead, we write this key to the `error_repo`, and `RGWDataSyncShardCR::incremental_sync()` will continue to retry it (see `/* process bucket shards that previously failed */`), while also making progress on the next bucket shards in the datalog

Reply:
Thank you for kindly review, to be honest, do not update the marker here has a lot of side effects, I pushed this commit here so that to discuss it with all of you, instead of this implement, maybe we can judge whether a shard is failed in `get_data_sync_status()` in rgw_admin.cc ?

Reply:
> do not update the marker here has a lot of side effects

i agree that it makes introspection more difficult - we might add some information to the `sync status` command about the number of entries in each shard's error_repo. are there other side effects that you worry about?

cc @yehudasa @oritwas

Reply:
get the failed shard_id and the number of entries in `error_repo` seems not difficult, but get the  `oldest change not applied` of a failed shard seems more difficult, hope all of you can give some suggestions on how to implement it in a appropriate way.
@cbodley @oritwas @yehudasa